### DISCUSSION TITLE
All my pdf bookmarks disappeared

### DISCUSSION THREAD

**Discussion by fluffy-marmot** [sumatrapdfreader/sumatrapdf#4040] [Documentation]
I've been using SumatraPDF for the past year or so and I've created quite a number of bookmarks\favorites to various PDFs, but seems like the last time I tried opening something to continue reading it, all my bookmarks, along with settings were gone (I usually use nightreading mode for white text on black background, but seems everything was back to default settings). Has anyone experienced a similar issue and knows how to recover the bookmarks \ settings by any chance?

**Comment by GitHubRulesOK** [comment]
unfortunatly there is no backup for recovery unless it was written  in your system so the slow system wide search is  from a CMD prompt (NOT the runbox, unless you just start cmd there) `where  /r c:\ sumatrapdf-settings.txt & pause` the pause ensures you later have time to see any results as it will be slow to search. currently I have loads from different portable copies !!

**Reply by fluffy-marmot** [reply]
where is the settings file normally stored? I don't delete random files for no reason, so I'm not sure why it would be gone...I've searched to see if others had similar issues and I found an old thread about a SumatraPDF update changing the location of the settings file and causing the same issue, so I'm wondering if this has happened to me as well, since I hadn't used it in a while and it's possible I updated it in the meantime.

**Reply by kjk** [reply]
See https://www.sumatrapdfreader.org/docs/How-we-store-settings

**Comment by fluffy-marmot** [comment]
It's happened again and I realized this time (and managed reliably duplicate it afterwards) that the favorites info gets deleted and all settings reset to factory settings if I have SumatraPDF open in the background and I shut down my laptop "improperly" by holding down the power button instead of exiting the program before shutting down my computer. I know it's preferable to use the proper Windows shut down procedure, but me that as it may, I don't really understand why Sumatra would be dependent upon writing these files on program exit...it doesn't seem like something the program should be vulnerable to, as comparable to having a file open in an editor and shutting it down while it's unsaved. Does anyone know why it behaves this way and whether this has something to do with my system settings or if it also behaves this way for others (if you are willing to risk losing all your favorites\settings...)

I like the simplicity of Sumatra and would like to continue to use it, but I really need to use favorites function to keep my place in long PDF books, and losing all of my saved places to every pdf I've ever opened - including ones that aren't open at the time! - just because my laptop has run out of battery or something, and I have Sumatra open in the background seems like a major design flaw and pain in the neck  :(

**Reply by GitHubRulesOK** [reply]
It is a weakness in many apps not just SumatraPDF Microsoft seem to have guarded against it in some but not all of their apps when the system does an enforced update with some files still open.
So in the past that was a frequent occurrence for "home" users. Which was improved in SumatraPDF but not  totally removed. If a file is fully open with edits and system has a "brownout" expect many files to be impacted. 
If you have extensive customisation it makes sense to backup any such files. So in this case the advanced settings file is the one major point of failure, For better or worse it is only one critical file for customisation and historic data , unless using multiple configurations ....

**Comment by kjk** [comment]
That shouldn't happen, I'll look into it. Consider opening an issue with this info so that it doesn't get lost.

**Comment by kjk** [comment]
Also, which version of sumatra are you using?

**Comment by kjk** [comment]
And is it portable or installed?

What is location of settings file on your computer?
